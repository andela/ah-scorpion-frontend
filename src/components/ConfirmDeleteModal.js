import React from 'react';
import Loader from './Loader';

const ConfirmDeleteModal = (props) => {
  const {
    confirmDelete,
    cancelDelete,
    deletedArticleSlug,
    isDeleting,
    errorMessage,
    deleteFailure,
    cleanDeleteFailure,
    deleteSuccess,
  } = props;

  return (
    <div
      className="modal fade"
      id="confirmDeleteModal"
      tabIndex="-1"
      role="dialog"
      data-backdrop="static"
    >
      <div className="modal-dialog" role="document">
        <div className="modal-content">
          <div className="modal-header">
            <h5
              className="modal-title mx-auto"
              id="confirmDeleteModalTilte"
            >
              Delete Article?
            </h5>
          </div>
          {isDeleting ? (
            <Loader
              style={{
                width: '20%',
                margin: 'auto',
              }}
            />
          ) : (
            <div className="modal-body">
              {deleteFailure || deleteSuccess ? (
                <div
                  className={`alert text-center alert-${deleteSuccess ? 'success' : 'danger'}`}
                >
                  {deleteFailure ? errorMessage : 'You have successfully deleted this article'}
                </div>
              ) : (
                <p className="text-center">
                Are you sure you want to delete this Article?
                  <br />
                This process is
                  {' '}
                  <strong className="text-danger">irreversible</strong>
.
                </p>
              )}
            </div>
          )}
          <div className="modal-footer">
            {deleteFailure || deleteSuccess ? (
              <button
                type="button"
                className="btn btn-secondary"
                data-dismiss="modal"
                disabled={isDeleting}
                onClick={cleanDeleteFailure}
              >
                Close
              </button>
            ) : (
              <button
                type="button"
                className="btn btn-secondary"
                data-dismiss="modal"
                disabled={isDeleting}
                onClick={cancelDelete}
              >
                Cancel
              </button>
            )}

            {deleteFailure || deleteSuccess ? (
              ''
            ) : (
              <button
                type="button"
                className="btn btn-danger"
                onClick={() => confirmDelete(deletedArticleSlug)}
                disabled={isDeleting}
              >
                {isDeleting ? 'Deleting...' : 'Delete'}
              </button>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};

export default ConfirmDeleteModal;
