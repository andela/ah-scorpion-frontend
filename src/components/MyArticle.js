import { Link } from 'react-router-dom';
import { Button } from 'react-bootstrap';
import PropTypes from 'prop-types';
import React from 'react';

const MyArticle = ({
  title,
  description,
  createdAt,
  slug,
  beginDelete,
  averageRating,
  likes,
  dislikes,
}) => (
  <div className="my-article">
    <h3>
      <Link to={`./article/${slug}`} className="article-title">
        {title}
      </Link>
    </h3>
    <p>{description}</p>
    <div>
      <span className="text-muted">
        <small>Created: {new Date(createdAt).toDateString()}</small>
      </span>
      <span className="ml-5">
        <Button bsStyle="primary">Edit</Button>
        <Button
          bsStyle="danger ml-2"
          data-toggle="modal"
          data-target="#confirmDeleteModal"
          onClick={() => beginDelete(slug)}
        >
          Delete
        </Button>
      </span>
    </div>
    <hr style={{ margin: '0.7em' }} />
    <div className="row">
      <div className="col-1 text-center">
        <i className="text-warning fa fa-star" />
        <br />
        <small>{averageRating || ' - '}</small>
      </div>
      <div className="col-1 text-center">
        <i className="text-success fa fa-thumbs-up" />
        <br />
        <small>{likes}</small>
      </div>
      <div className="col-1 text-center">
        <i className="text-danger fa fa-thumbs-down" />
        <br />
        <small>{dislikes}</small>
      </div>
    </div>
  </div>
);

MyArticle.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
  averageRating: PropTypes.number,
  likes: PropTypes.number.isRequired,
  dislikes: PropTypes.number.isRequired,
};

MyArticle.defaultProps = {
  averageRating: null,
};

export default MyArticle;
