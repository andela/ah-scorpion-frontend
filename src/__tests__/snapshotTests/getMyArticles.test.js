import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import ConfirmDeleteModal from '../../components/ConfirmDeleteModal';
import MyArticle from '../../components/MyArticle';
import MyArticleList from '../../components/MyArticlesList';

describe('The getmyArticlesPage view', () => {
  describe('The ConfirmDeleteModal component', () => {
    const state = {
      confirmDelete: () => 1,
      cancelDelete: () => 1,
      deletedArticleSlug: 'test article',
      isDeleting: false,
      errorMessage: '',
      deleteFailure: false,
      postRequestCleanUp: () => 1,
      deleteSuccess: false,
    };
    it('Should initially ask for delete confirmation', () => {
      const tree = renderer.create(<ConfirmDeleteModal {...state} />);
      expect(tree.toJSON()).toMatchSnapshot();
    });
    it('Should show loading status when request is pending', () => {
      const tree = renderer.create(<ConfirmDeleteModal {...state} isDeleting />);
      expect(tree.toJSON()).toMatchSnapshot();
    });
    it('Should show success status when request is successful', () => {
      const tree = renderer.create(<ConfirmDeleteModal {...state} deleteSuccess />);
      expect(tree.toJSON()).toMatchSnapshot();
    });
    it('Should show error status when request fails', () => {
      const tree = renderer.create(<ConfirmDeleteModal {...state} deleteFailure />);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
  describe('The MyArticle component', () => {
    it('Should not regress', () => {
      const tree = renderer.create(
        <MemoryRouter>
          <MyArticle
            title="test title"
            description="test description"
            createdAt="2018-09-12T18:51:24.099861Z"
            slug="test-slug-001"
            beginDelete={() => 1}
          />
        </MemoryRouter>,
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe('The MyArticleList component', () => {
    const testProps = {
      articles: [],
      beginDelete: () => 1,
      fetchFailure: false,
      errorMessage: '',
    };
    const articles = [
      {
        id: 43,
        title: "Lenny's First Article 1",
        author: {
          email: 'lennykmutua@gmail.com',
          username: 'lenny',
          bio: '',
          image: '',
        },
        likes: 0,
        dislikes: 0,
        favorited: 1,
        averageRating: null,
        ratingsCount: 0,
        slug: 'lennys-first-article-1-29b5102ef9a84eb9bfbf60da42f63122',
        body: "Like I said... I don't know what's going on. I'm just testing Hoslack's work.",
        description: "This is my first article and I don't know what's going on...",
        images: ['https://url-to-image'],
        createdAt: '2018-09-12T18:51:24.099861Z',
        updatedAt: '2018-09-12T18:51:24.100977Z',
        tagList: ['Lenny', 'First'],
      },
      {
        id: 43,
        title: "Lenny's First Article 2",
        author: {
          email: 'lennykmutua@gmail.com',
          username: 'lenny',
          bio: '',
          image: '',
        },
        likes: 0,
        dislikes: 0,
        favorited: 1,
        averageRating: null,
        ratingsCount: 0,
        slug: 'lennys-first-article-2-29b5102ef9a84eb9bfbf60da42f63122',
        body: "Like I said... I don't know what's going on. I'm just testing Hoslack's work.",
        description: "This is my first article and I don't know what's going on...",
        images: ['https://url-to-image'],
        createdAt: '2018-09-12T18:51:24.099861Z',
        updatedAt: '2018-09-12T18:51:24.100977Z',
        tagList: ['Lenny', 'First'],
      },
    ];
    it('Should render text if user has no articles', () => {
      const tree = renderer.create(
        <MemoryRouter>
          <MyArticleList {...testProps} />
        </MemoryRouter>,
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
    it('Should render the error message in case of failure', () => {
      const tree = renderer.create(
        <MemoryRouter>
          <MyArticleList {...testProps} fetchFailure errorMessage="An error occurred" />
        </MemoryRouter>,
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });

    it('Should render all articles in case of a successful request', () => {
      const tree = renderer.create(
        <MemoryRouter>
          <MyArticleList {...testProps} articles={articles} />
        </MemoryRouter>,
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
