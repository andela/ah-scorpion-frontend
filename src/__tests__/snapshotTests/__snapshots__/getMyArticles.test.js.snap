// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The getmyArticlesPage view The ConfirmDeleteModal component Should initially ask for delete confirmation 1`] = `
<div
  className="modal fade"
  data-backdrop="static"
  data-keyboard="false"
  id="confirmDeleteModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title mx-auto"
          id="confirmDeleteModalTilte"
        >
          Delete Article?
        </h5>
      </div>
      <div
        className="modal-body"
      >
        <p
          className="text-center"
        >
          Are you sure you want to delete this Article?
          <br />
          This process is
           
          <strong
            className="text-danger"
          >
            irreversible
          </strong>
          .
        </p>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <button
          className="btn btn-danger"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`The getmyArticlesPage view The ConfirmDeleteModal component Should show error status when request fails 1`] = `
<div
  className="modal fade"
  data-backdrop="static"
  data-keyboard="false"
  id="confirmDeleteModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title mx-auto"
          id="confirmDeleteModalTilte"
        >
          Delete Article?
        </h5>
      </div>
      <div
        className="modal-body"
      >
        <div
          className="alert text-center alert-danger"
        >
          
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Close
        </button>
        
      </div>
    </div>
  </div>
</div>
`;

exports[`The getmyArticlesPage view The ConfirmDeleteModal component Should show loading status when request is pending 1`] = `
<div
  className="modal fade"
  data-backdrop="static"
  data-keyboard="false"
  id="confirmDeleteModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title mx-auto"
          id="confirmDeleteModalTilte"
        >
          Delete Article?
        </h5>
      </div>
      <img
        alt="Loading..."
        src={
          Object {
            "0": "l",
            "1": "o",
            "2": "a",
            "3": "d",
            "4": "e",
            "5": "r",
            "6": ".",
            "7": "s",
            "8": "v",
            "9": "g",
            "default": "loader.svg",
          }
        }
        style={
          Object {
            "margin": "auto",
            "width": "20%",
          }
        }
      />
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <button
          className="btn btn-danger"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Deleting...
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`The getmyArticlesPage view The ConfirmDeleteModal component Should show success status when request is successful 1`] = `
<div
  className="modal fade"
  data-backdrop="static"
  data-keyboard="false"
  id="confirmDeleteModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title mx-auto"
          id="confirmDeleteModalTilte"
        >
          Delete Article?
        </h5>
      </div>
      <div
        className="modal-body"
      >
        <div
          className="alert text-center alert-success"
        >
          You have successfully deleted this article
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Close
        </button>
        
      </div>
    </div>
  </div>
</div>
`;

exports[`The getmyArticlesPage view The MyArticle component Should not regress 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "testKey",
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <MyArticle
    averageRating={null}
    beginDelete={[Function]}
    createdAt="2018-09-12T18:51:24.099861Z"
    description="test description"
    slug="test-slug-001"
    title="test title"
  />
  <ReadTime
    text="This is a test text"
  />
</Router>
`;

exports[`The getmyArticlesPage view The MyArticleList component Should render all articles in case of a successful request 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "testKey",
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <MyArticlesList
    articles={
      Array [
        Object {
          "author": Object {
            "bio": "",
            "email": "lennykmutua@gmail.com",
            "image": "",
            "username": "lenny",
          },
          "averageRating": null,
          "body": "Like I said... I don't know what's going on. I'm just testing Hoslack's work.",
          "createdAt": "2018-09-12T18:51:24.099861Z",
          "description": "This is my first article and I don't know what's going on...",
          "dislikes": 0,
          "favorited": 1,
          "id": 43,
          "images": Array [
            "https://url-to-image",
          ],
          "likes": 0,
          "ratingsCount": 0,
          "slug": "lennys-first-article-1-29b5102ef9a84eb9bfbf60da42f63122",
          "tagList": Array [
            "Lenny",
            "First",
          ],
          "title": "Lenny's First Article 1",
          "updatedAt": "2018-09-12T18:51:24.100977Z",
        },
      ]
    }
    beginDelete={[Function]}
    errorMessage=""
    fetchFailure={false}
  />
</Router>
`;

exports[`The getmyArticlesPage view The MyArticleList component Should render text if user has no articles 1`] = `
<h4>
  You have not created any articles yet.
</h4>
`;

exports[`The getmyArticlesPage view The MyArticleList component Should render the error message in case of failure 1`] = `
<div
  className="alert alert-danger text-center"
>
  An error occurred
</div>
`;

exports[`The getmyArticlesPage view The ReadTime component Should not regress 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "testKey",
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <ReadTime
    text="This is a test text"
  />
</Router>
`;
